### ISSUE TITLE
Feature-gate accessibility in `bevy_ui`

### ISSUE THREAD

**Issue by Niashi24** [bevyengine/bevy#16312]
## What problem does this solve or what need does it fill?

Due to `bevy_ui`'s dependency on `accesskit` (and `bevy_a11y`) which depends on `std`, `bevy_ui` cannot currently support `no_std`. However, UI is foundational to many games, and other features of `bevy_ui` that would not depend on `std` (especially layouting) would be great to have in `no_std`. While accessibility features may be valuable to have in `no_std`, we are currently blocked on the ecosystem. In the meantime, these features should be feature-gated out of `bevy_ui` to allow for future `no_std` support.

## What solution would you like?

Add an `accessibility` feature and gate integration with `accesskit` and `bevy_a11y` behind it. This should be enabled by default.

## What alternative(s) have you considered?

Break out the base UI Node and layouting systems into a separate crate (`bevy_layout`?) that could then be made `no_std` on its own. This could be confusing for people but also might be worth it if `bevy_ui` gets too big in the future.

## Additional context

`no_std` support for `bevy_ui` is also currently blocked on `no_std` support for `bevy_ecs` and `bevy_app` (#15460), and a dependency with `bevy_render`. These can be done in parallel however.

**Comment by DataTriny** [comment]
FYI the core of AccessKit, that is the accesskit crate, is no-std compatible now. Since the other adapter crates are meant to be compiled only on the relevant platforms, bevy_ui can keep AccessKit support and still be no-std compatible without a feature.
